modules/tensor_mechanics/src/materials/ComputeDilatationThermalExpansionFunctionEigenstrain.C

```diff
@@ -36,19 +36,16 @@ ComputeDilatationThermalExpansionFunctionEigenstrainTempl<is_ad>::
 }
 
 template <bool is_ad>
-GenericReal<is_ad>
+ValueAndDerivative<is_ad>
```

Comment:
Are we potentially making copies of `ADReal`s by having the return types without references?

Reply:
RVO or NRVO should prevent the copying, and C++17 guarantees copy elision.